File: 1101-1200/1103-Distribute Candies to People/1103-Distribute Candies to People.py
class Solution:
    def distributeCandies(self, candies: int, num_people: int) -> List[int]:
        result = [0] * num_people
        curr = 1
        while candies > 0:
            for i in range(num_people):
                if candies >= curr + i:
                    result[i] += curr + i
                    candies -= curr + i
                else:
                    result[i] += candies
                    candies = 0
                    break
            curr += num_people

        return result


class Solution2:
    def distributeCandies(self, candies: int, num_people: int) -> List[int]:
        result = [0] * num_people
        curr = 0
        while candies > 0:
            result[curr % num_people] += min(candies, curr + 1)
            curr += 1
            candies -= curr

        return result

File: 1101-1200/1103-Distribute Candies to People/1103-Distribute Candies to People.cpp
class Solution {
public:
    vector<int> distributeCandies(int candies, int num_people) {
        vector<int> result(num_people);
        int curr = 1;
        while (candies > 0) {
            for (int i = 0; i < num_people; ++i) {
                if (candies >= curr + i) {
                    result[i] += curr + i;
                    candies -= curr + i;
                }
                else {
                    result[i] += candies;
                    candies = 0;
                    break;
                }
            }

            curr += num_people;
        }

        return result;
    }
};
